packages/metro/src/DeltaBundler/__tests__/traverseDependencies-test.js

```diff
@@ -681,6 +693,408 @@ describe('edge cases', () => {
     ]);
   });
 
+  describe('adding a new module while one of its ancestors is being deleted', () => {
+    /**
+     * In all of these tests, we start with the following graph:
+     *
+     * ┌─────────┐     ┌──────┐     ┌──────┐
+     * │ /bundle │ ──▶ │ /foo │ ──▶ │ /bar │
+     * └─────────┘     └──────┘     └──────┘
+     *                   │
+     *                   │
+     *                   ▼
+     *                 ┌──────┐
+     *                 │ /baz │
+     *                 └──────┘
+     *
+     * and end up with the following graph:
+     *
+     * ┌─────────┐     ┌───────┐
+     * │ /bundle │ ──▶ │ /quux │
+     * └─────────┘     └───────┘
+     *
+     * In the process, we add a "live" edge from /bundle to /quux (which makes it to the final
+     * graph) and a "dead" edge from /foo to /quux (which does not, because /foo is pruned).
+     *
+     * The order and chunking of mutations should not affect the correctness of the graph
+     * (but has been known to cause bugs, hence these regression tests).
+     */
+    it('all in one delta, adding the live edge first', async () => {
+      await initialTraverseDependencies(graph, options);
+
+      /*
+      ┌─────────┐     ┌──────┐     ┌──────┐
+      │ /bundle │ ──▶ │ /foo │ ──▶ │ /bar │
+      └─────────┘     └──────┘     └──────┘
+                        │
+                        │
+                        ▼
+      ┏━━━━━━━━━┓     ┌──────┐
+      ┃  /quux  ┃     │ /baz │
+      ┗━━━━━━━━━┛     └──────┘
+      */
+      Actions.createFile('/quux');
+
+      /*
+      ┌─────────┐     ┌──────┐     ┌──────┐
+      │ /bundle │ ──▶ │ /foo │ ──▶ │ /bar │
+      └─────────┘     └──────┘     └──────┘
+        ┃	              │
+        ┃               │
+        ▼               ▼
+      ┌─────────┐     ┌──────┐
+      │  /quux  │     │ /baz │
+      └─────────┘     └──────┘
+      */
+      Actions.addDependency('/bundle', '/quux');
+
+      /*
```

Comment:
This tests look very similar to the ones I illustrated here -- https://github.com/facebook/metro/issues/500